crates/bevy_asset/src/handle.rs

```diff
@@ -138,6 +138,22 @@ pub enum Handle<A: Asset> {
     Weak(AssetId<A>),
 }
 
+/// Serialize `Handle`s as [`AssetPath`]s. This means types containing handles cannot be
+/// deserialized without special handling. See
+/// [`ReflectDeserializerProcessor`](bevy_reflect::serde::ReflectDeserializerProcessor).
+impl<A: Asset> serde::Serialize for Handle<A> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match self.path() {
+            Some(asset_path) => asset_path.serialize(serializer),
+            // Maybe this should be an error?
+            None => AssetPath::default().serialize(serializer),
```

Comment:
This definitely feels like it should be an error case: this will cause serious bugs if people don't catch this.

Reply:
I agree with this. I think we can return an error with `S::Error::custom("blah")`.